Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -11591,9 +11590,86 @@ void CPhysicalGeometry::SetWallDistance(const CConfig *config, CADTElemClass *Wa
       unsigned long  elemID;
       int            rankID;
       su2double      dist;
+      su2double      localRoughness;
+
       WallADT->DetermineNearestElement(nodes->GetCoord(iPoint), dist, markerID, elemID, rankID);
+
       nodes->SetWall_Distance(iPoint, min(dist,nodes->GetWall_Distance(iPoint)));
+      localRoughness = config->GetLocalRoughness(rankID, markerID);
+      nodes->SetRoughnessHeight(iPoint, localRoughness);
     }
+    
   }
   // end SU2_OMP_PARALLEL
 }
+
+void CPhysicalGeometry::SetGlobalMarkerRoughness(CConfig *config) {
+
+unsigned short iMarker;
+#ifndef HAVE_MPI
+  
+  int *local_displ = new int [1];
+  int size = 1;
+  unsigned short nMarker_All = config->GetnMarker_All();
+  local_displ[0] = 0;
+  config->SetGlobalMarkerArray(local_displ,size);
+  
+  su2double *localRough = new su2double [nMarker_All];
+
+  if (config->GetnRoughWall() > 0)
+    for (iMarker = 0; iMarker < nMarker_All; iMarker++)
+      localRough[iMarker] = config->GetWall_RoughnessHeight(config->GetMarker_All_TagBound(iMarker));   
+  else 
+    for (iMarker = 0; iMarker < nMarker_All; iMarker++) 
+      localRough[iMarker] = 0.0 ;
+
+  config->SetGlobalRoughnessArray(localRough, nMarker_All);
+  
+#else
+  int rank, size;
+  unsigned short nMarker_All = config->GetnMarker_All();
+  SU2_MPI::Comm_rank(MPI_COMM_WORLD, &rank);
+  SU2_MPI::Comm_size(MPI_COMM_WORLD, &size);
+
+  int *displs = new int [size];
+  int* recvCounts = new int [size];//, displs(size);
+  int sizeLocal = (int) nMarker_All; // number of local markers
+  
+  //Communicate size of local marker array and make an array large enough to hold all data
+  SU2_MPI::Allgather(&sizeLocal, 1, MPI_INT, recvCounts, 1,
+                       MPI_INT, MPI_COMM_WORLD);
+  
+  SU2_MPI::Barrier(MPI_COMM_WORLD);   // Not sure if needed
```

Comment:
This routine is perhaps not very efficient and I appreciate any comments to improve it.